examples/carplay-web-app/src/worker/Render.worker.ts

```diff
@@ -0,0 +1,192 @@
+// Based on https://github.com/codewithpassion/foxglove-studio-h264-extension/tree/main
+// MIT License
+import { NALUStream, SPS } from '../lib/h264-utils'
+import {
+  getNalus,
+  identifyNaluStreamInfo,
+  NaluStreamInfo,
+  NaluTypes,
+} from '../lib/utils'
+import {
+  InitRenderEvent,
+  RenderDomeEvent as RenderDoneEvent,
+  RenderEvent,
+  StatusEvent,
+  WorkerEvent,
+} from './RenderEvents'
+import { WebGLRenderer } from './WebGLRenderer'
+
+export type StatusType = 'render' | 'decode'
+type PartialRecord<K extends string, T> = Partial<Record<K, T>>
+export type StatusUpdate = PartialRecord<StatusType, string>
+
+// eslint-disable-next-line no-restricted-globals
+const scope = self as unknown as Worker
+
+type HostType = Window & typeof globalThis
+
+export class RenderWorker {
```

Comment:
There's quite a few usages of `.bind(this)` in this class, they can be eliminated by defining the class members in an alternative way, ie

```diff
-   private onVideoDecoderOutput(frame: VideoFrame) {
+   private onVideoDecoderOutput = (frame: VideoFrame) => {
```